tool/gravity/cli/operation.go

```diff
@@ -210,27 +231,17 @@ func (r *backendOperations) List(localEnv, updateEnv, joinEnv *localenv.LocalEnv
 			log.WithError(err).Warn("Failed to query cluster operations.")
 		}
 	}
-	if updateEnv != nil {
-		r.getOperationAndUpdateCache(updateEnv.Backend, log.WithField("context", "update"))
+	if err := r.listUpdateOperation(environ); err != nil && !trace.IsNotFound(err) {
```

Comment:
Hm, I might be misunderstanding the logic here, but the way it was implemented before IIRC we would pass either updateEnv or joinEnv here - one of those would be nil depending on which command is run - and hence we would only query either update or join operation. Now, we query both so backendOperations may contain both upgrade and join operation if both backends are present on the node. Would that confuse anything?

Reply:
No, prior to that both `updateEnv` and `joinEnv` could be active at the same time as the `plan` command is ambiguous so it worked similar to this before. 
One thing that to account for is the residual state that might have been left behind as the `leave` (or more specifically `system uninstall`) command would not remove the join state directory (only relevant in 5.5.x though) and since the expand operation was never updated to completion in this local store, it will popup as an unfinished (hence active) and maybe even unrelated to the current cluster.
All because of such disparate state management.